robotiq_joint_state_publisher/src/s_model_joint_states.cpp

```diff
@@ -0,0 +1,208 @@
+//  The MIT License (MIT)
+//
+//  Copyright (c) 2014 Jack Thompson
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+/** @file s_model_joint_states.cpp
+ *  Subscribes to Robotiq state messages on "SModelRobotInput" topic, converts the data to joint values,
+ *  and publishes sensor_msgs/JointState messages on "joint_states" topic for Robotiq S-model.
+ * 
+ *  'rosrun robotiq_joint_state_publisher s_model_joint_states <gripper_prefix>'
+ * 
+ *  @author jack.thompson(at)utexas.edu
+ *  @maintainer karl.kruusamae(at)utexas.edu
+ */
+
+#include <vector>
+#include <string>
+#include <csignal>
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include <robotiq_s_model_control/SModel_robot_input.h>
+#include <robotiq_s_model_control/SModel_robot_output.h>
+
+
+const int BASIC   = 0;
+const int PINCH   = 1;
+const int WIDE    = 2;
+const int SCISSOR = 3;
+
+const double DEG_TO_RAD = M_PI/180.0;
+
+/**
+ * Finger class for single Robotiq S-model finger.
+ */
+class Finger {
+ public:
+  Finger() { position = 0; }				///< Default constructor for creating Finger, position is set 0
+  Finger(int pos) { position = pos; }			///< Create Finger, position is @param pos
+  Finger(const Finger &f) { position = f.position; }	///< Create Finger, position is taken form @param Finger
+  inline double joint_1() const;			///< joint_1 value for Finger
+  inline double joint_2() const;			///< joint_2 value for Finger
+  inline double joint_3() const;			///< joint_3 value for Finger
+  int position;						///< Position of the Finger
+};
+
```

Comment:
I'm not familiar with the S-model kinematics/interface.  Can you explain (in a comment) how the joint angles are derived from a single `position`?  I assume these calculation are only correct in a non-interfering state.

There's also a few too many magic numbers for me.  I don't know that making named constants really helps here, since the values seem arbitrary.  Is there a meaning behind the repeated numbers, like `70` and `140`